include/pybind11/pybind11.h

```diff
@@ -581,7 +581,7 @@ class cpp_function : public function {
                         extra_args = tuple(args_size);
                         for (size_t i = 0; i < args_size; ++i) {
                             handle item = PyTuple_GET_ITEM(args_in, args_copied + i);
-                            extra_args[i] = item.inc_ref().ptr();
+                            extra_args[i] = reinterpret_borrow<object>(item);
```

Comment:
The `reinterpret_borrow` and the temporary variable aren't actually needed; the two lines can be simplified to just:
```C++
extra_args[i] = PyTuple_GET_ITEM(args_in, args_copied + i);
```